src/lib/classes/engine/Engine.ts

```diff
@@ -0,0 +1,129 @@
+import { EngineType } from "./EngineType";
+import {
+	comparePortRange,
+	validateEndPort,
+	validateIP,
+	validateName,
+	validateStartPort,
+} from "./Validation";
+
+/**
+ * A Reveaal, JEcdar andd API engine definition
+ * It stores the IP and port of an engine
+ * */
+export class Engine {
+	/**
+	 * The #name of the Engine
+	 * */
+	#name: string = "";
+	get name(): string {
+		return this.#name;
+	}
+	set name(setName: string | undefined) {
+		if (validateName(setName) && setName !== undefined)
```

Comment:
```suggestion
		if (validateName(setName))
```
This check is already performed in `validateName`